bin/express-cli.js

```diff
@@ -93,9 +94,10 @@ function createApplication (name, dir, options, done) {
   var pkg = {
     name: name,
     version: '0.0.0',
+    type: options.es6 ? 'module' : 'commonjs',
```

Comment:
Have you considered using a conditional and adding the type property instead of defaulting to `commonjs` ? The current code doesn't set a type property and defaulting to `commonjs` has a potential to change behavior for existing code.

```js
  var pkg = {
    name: name,
    version: '0.0.0',
    private: true,
    scripts: {
      start: options.esm ? 'node ./bin/www.js' : 'node ./bin/www'
    },
    dependencies: {
      debug: '~2.6.9',
      express: '~4.17.1'
    }
  }
  if (options.esm) {
    pkg.type = 'module'
  }
```

Reply:
Yes, I considered not setting type: commonjs explicitly.  I'm almost positive that this will not change existing code.  For one thing, express-generator is generating new code ;-)  For another, even if someone dropped existing code underneath the generated code, the behavior of the existing code should not change since commonjs is the default for type.  All this line does is make this default explicit and "future-proof" the package against possible later changes to the default.  That's a good thing, I think.

Reply:
I see your point about commonjs being the default module loading style for Node. I’m just going to call this out as a risky change because it’s technically impacting existing functionality and can be written in a way to avoid that risk. Ultimately, only time can tell what the positive move is.

Reply:
✔️ Fair enough.

Reply:
node isn’t going to be able to change the default; it’d break the world.

Reply:
I wouldn't be too sure of that. After all, many folks thought that Y2K bugs might literally break the world, but they didn't.

Reply:
LOL. I heard someone say that Y2K bugs didn't break the world because of all the planning and work to prepare for it.

Reply:
True enough. But in the case of changing the type default, the work for an individual package is adding one line to package.json if it's not already there. One could even imagine automated fixes for this change as opposed to bringing COBOL programmers out of retirement ;-)